hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -41,6 +42,13 @@
      */
     boolean hasPartitionIndexed(int partitionId);
 
+    /**
+     * Returns {@code true} if all {@code queryPartitions} are indexed,
+     * {@code false} otherwise.
+     * @param queryPartitions a set of partitions a query runs on.
+     */
+    boolean allPartitionsIndexed(PartitionIdSet queryPartitions);
```

Comment:
I looked at two implementations of this method (non-HD and HD), and it seems that none of them needs `PartitionIdSet`. Non-HD implementation compares the number of currently indexed partitions with the number of currently owned partitions. HD implementation doesn't need `PartitionIdSet` at all. It seems that the whole interface could be simplified to `allPartitionsIndexed()` (together with a significant part of the patch), if we allow `IndexImpl` to get the number of currently owned partitions internally.

Reply:
The trick is that we don't have a reference to the member from `IndexImpl` and we don't want to add this extra dependency. This is why we pass the `queryPartitions` deep inside.